src/mlpack/methods/logistic_regression/logistic_regression.hpp

```diff
@@ -53,6 +53,30 @@ class LogisticRegression
   LogisticRegression(const MatType& predictors,
                      const arma::Row<size_t>& responses,
                      const double lambda = 0);
+  /**
+   * Construct the LogisticRegression class with the given labeled training
+   * data.  This will train the model.  Optionally, specify lambda, which is the
+   * penalty parameter for L2-regularization.  If not specified, it is set to 0,
+   * which results in standard (unregularized) logistic regression.
+   *
+   * It is not possible to set a custom optimizer with this constructor.  Either
+   * use a constructor that does not train and call Train() with a custom
+   * optimizer type, or use the constructor that takes an instantiated
+   * optimizer.  (This unfortunate situation is a language restriction of C++.)
+   *
+   * This constructor is necessary for the cross-validation code and
+   * hyper-parameter tuner to work, but it isn't suggested for regular use since
+   * it requires the extra numClasses parameter (which must always be 2).
+   *
+   * @param predictors Input training variables.
+   * @param responses Outputs resulting from input training variables.
+   * @param numClasses Number of classes in the dataset (must be 2).
+   * @param lambda L2-regularization parameter.
+   */
+  LogisticRegression(const MatType& predictors,
+                     const arma::Row<size_t>& responses,
+                     const size_t numClasses,
```

Comment:
Hm, so I spent some time thinking about this point.  I think that in order to preserve compatibility of two-class classifiers with other classifiers, we can simply provide two overloads, one that takes `numClasses` (so that the classifier satisfies the requirements of other classifiers) and one that doesn't, for user convenience.

AdaBoost is a multiclass classifier (or it can be, depending on its weak learner) so I think leaving `numClasses` there is just fine.

Do you think that is ok, or does that cause any problems?